src/pmpo_MPMesh.cpp

```diff
@@ -258,17 +296,52 @@ void MPMesh::T2LTracking(Vec2dView dx){
     p_MPs->parallel_for(T2LCalc,"T2lTrackingCalc");
 }
 
+void MPMesh::reconstructSlices() {
+    if (reconstructSlice.size() == 0) return;
+    Kokkos::Timer timer;
+    calcBasis();
+    for (auto const& [index, reconstruct] : reconstructSlice) {
+        if (reconstruct) reconstruct();
+    }
+    reconstructSlice.clear();
+    pumipic::RecordTime("PolyMPO_Reconstruct", timer.seconds());
+}
+
+bool getAnyIsMigrating(bool isMigrating) {
+  Kokkos::Timer timer;
+  int comm_rank;
+  MPI_Comm_rank(MPI_COMM_WORLD, &comm_rank);
+  int comm_size;
+  MPI_Comm_size(MPI_COMM_WORLD, &comm_size);
```

Comment:
Need to completely avoid MPI_COMM_WORLD communicator. And need to use application provided communicator.